applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py

```diff
@@ -212,15 +211,30 @@ def Initialize(self):
 
         # Initialize the distance correction process
         self._GetDistanceModificationProcess().ExecuteInitialize()
-
+        self._GetDistanceModificationProcess().ExecuteInitializeSolutionStep()
+        
+        #Here water volume of the system is calculated considering inlet and outlet flow rate. Qsystem= Q_inlet-Q_oulet+Q_system(previous_dt)
+        self.initial_system_volume=KratosCFD.FluidAuxiliaryUtilities.CalculateFluidNegativeVolume(self.GetComputingModelPart())
+    
         # Instantiate the level set convection process
         # Note that is is required to do this in here in order to validate the defaults and set the corresponding distance gradient flag
         # Note that the nodal gradient of the distance is required either for the eulerian BFECC limiter or by the algebraic element antidiffusivity
         self._GetLevelSetConvectionProcess()
 
         KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Solver initialization finished.")
-
+        
     def InitializeSolutionStep(self):
+
+        # Inlet and outlet water discharge is calculated for current time step, first discharge and the considering the time step inlet and outlet volume is calculated
+        self.outlet_discharge=KratosCFD.FluidAuxiliaryUtilities.CalculateFlowRateNegativeSkin(self.GetComputingModelPart(),KratosMultiphysics.OUTLET)
+        self.inlet_discharge=KratosCFD.FluidAuxiliaryUtilities.CalculateFlowRateNegativeSkin(self.GetComputingModelPart(),KratosMultiphysics.INLET)
+        self.current_dt=self.main_model_part.ProcessInfo[KratosMultiphysics.DELTA_TIME]
+        self.inlet_volume=-self.current_dt*self.inlet_discharge
+        self.outlet_volume=self.current_dt*self.outlet_discharge
+
+        # System water volume is calculated for current time step considering inlet and outlet discharge. 
+        self.system_volume= self.inlet_volume+self.initial_system_volume-self.outlet_volume
```

Comment:
```suggestion
        outlet_discharge = KratosCFD.FluidAuxiliaryUtilities.CalculateFlowRateNegativeSkin(self.GetComputingModelPart(),KratosMultiphysics.OUTLET)
        inlet_discharge = KratosCFD.FluidAuxiliaryUtilities.CalculateFlowRateNegativeSkin(self.GetComputingModelPart(),KratosMultiphysics.INLET)
        current_dt = self.main_model_part.ProcessInfo[KratosMultiphysics.DELTA_TIME]
        inlet_volume = -current_dt * inlet_discharge
        outlet_volume = current_dt * outlet_discharge

        # System water volume is calculated for current time step considering inlet and outlet discharge. 
        system_volume = inlet_volume + initial_system_volume - outlet_volume
```
You don't require these to be member variables.